### PR TITLE
Safety rails on decode

### PR DESCRIPTION
**Pull request by jordanmessina**

### CODE REVIEW

File: src/protocol/weth/weth.ts [hunk 1]
```diff
@@ -21,16 +21,16 @@ export const detectWeth = (transaction: Transaction): boolean => {
...
+      decode = decodeTransactionInput(transaction.input, WETH_ABI);
+    } catch (e) {
+      return false;
+    }
 
-    if (
-      transactionDescriptor.name !== 'deposit' &&
-      transactionDescriptor.name !== 'withdraw'
-    ) {
+    if (decode.name !== 'deposit' && decode.name !== 'withdraw') {
```
Comment (pcowgill) [inline src/protocol/weth/weth.ts:33]:
```suggestion
    if (!decode || !decode.name) return false;
    if (decode.name !== 'deposit' && decode.name !== 'withdraw') {
```

File: src/protocol/weth/weth.ts [hunk 2]
```diff
@@ -101,14 +95,8 @@ export const generateWethContext = (transaction: Transaction): Transaction => {
...
         variables: {
-          value: {
-            type: 'eth',
-            value: ethFlow?.[0].value,
-          },
-          from: {
-            type: 'address',
-            value: transaction.from,
-          },
+          withdraw,
```
Comment (pcowgill) [inline src/protocol/weth/weth.ts:98]: This variable name is less clear than value. Maybe `withdrawalAmount`, if you don't like `value`?

File: src/protocol/weth/weth.ts [hunk 3]
```diff
@@ -77,18 +74,15 @@ export const generateWethContext = (transaction: Transaction): Transaction => {
...
-      }
-
+      const decode = decodeTransactionInput(transaction.input, WETH_ABI);
+      const withdrawer: ContextSummaryVariableType = {
+        type: 'address',
+        value: transaction.from,
+      };
+      const withdraw: ContextSummaryVariableType = {
+        type: 'eth',
+        value: decode.args[0],
```
Comment (pcowgill) [inline src/protocol/weth/weth.ts:85]: Ah yeah, this is a more direct way to get it as opposed to using netAssetTransfers

**Comment by pcowgill** [comment]
Demonstrating that the contextualizers all run reliably with these changes
Screenshot 2023-12-04 at 10 22 00 PM

File: src/protocol/weth/weth.ts (hunk 3 above)
Comment (jordanmessina) [reply src/protocol/weth/weth.ts:85]: netAssetTransfers is great for heuristics or protocol contextualizations that have gnarly input data. But when we can simply use the abi, we should.

Comment (pcowgill) [reply src/protocol/weth/weth.ts:85]: Agreed!

File: src/protocol/weth/weth.ts (hunk 2 above)
Comment (pcowgill) [reply src/protocol/weth/weth.ts:99]: Let's update .en.default to reflect these changes

**Review by pcowgill — APPROVED** [review]

**Pull request merged**